tests/test_aird_parser.py

```diff
@@ -30,6 +31,13 @@ def diagram_under_test(self, model):
         assert descriptor is not None
         yield aird.parse_diagram(model, descriptor)
 
+    def test_yield_diagram_elements(self, model):
+        diagram_elements = list(
+            aird.yield_diagram_elements(model, self.test_diagram_uid)
+        )
+
+        assert len(diagram_elements) == 16
```

Comment:
This can easily (and should) be parameterized for all currently existing diagrams in the 5.0 model. This would have already found the discrepancy with `master` I pointed out in a different comment; it just so happens that this particular diagram is not affected.

To use the actual 5.0 test model, this also needs to be moved out of this class.

Here you go:

```python
@pytest.mark.parametrize(
    ["diag_uid", "num_nodes"],
    [
        ("_7FWu4KrxEeqOgqWuHJrXFA", 34),
        ("_KK2wcKyJEeqCdMaqCWkrKg", 14),
        ("_beibwNYrEeqiU8uzTY0Puw", 11),
        ("_9inmIKgWEeujco-rU7ZOtA", 1),
        ("_Go81cDaWEey5m4gkio8hew", 2),
        ("_-NRNIF2mEey8erljvkS-pQ", 13),
        ("_KZOz4GA9Eey8erljvkS-pQ", 15),
        ("_joc_sGDdEey8erljvkS-pQ", 5),
        ("_bpsckGDhEey8erljvkS-pQ", 5),
        ("_OAg8QHPIEeyW3OIB4qRWZA", 18),
        ("_RSWgcHPIEeyW3OIB4qRWZA", 8),
        ("_bNLx4HPIEeyW3OIB4qRWZA", 6),
        ("_sB8k8Mn-EeyS2Zr7ZWFrXA", 7),
        ("_1adB8GpbEe2Scdb2k-Fvhg", 25),
        ("_fqzi8BKnEeuBCogvtwwNBw", 78),
        ("_ervhAFIqEeyiRNlyKPJwqw", 60),
        ("_8BMtsIQzEeyxv9w6U6_UQg", 10),
        ("_al3zwKg0EeujRPrkuugYGw", 5),
        ("_1sCb8OSREeulYvEGRaHazg", 8),
        ("_APMboAPhEeynfbzU12yy7w", 11),
        ("_K9160IKBEeyd4IW8P6K-AQ", 27),
        ("_0g-esB4cEe2Jgqeci1tAhA", 1),
        ("_hIt8QKWXEeqh-bahrg4jkA", 18),
        ("_u8lIgLB-EeqP7JXhmLvOsg", 26),
        ("_2SjcgAL8EeuogvvqBpYPnQ", 11),
        ("_KTxZ8KJwEeusg-blS2y7tg", 24),
        ("_ZNrYAKJxEeusg-blS2y7tg", 31),
        ("_mKI_kKZwEeuRO5WMG4dh5w", 4),
        ("_MDoM4AWrEeyIAukYgdUX5A", 11),
        ("_VKPRsIQTEeyxv9w6U6_UQg", 26),
        ("_dpUi4LB8EeqLtsTDMgEHBw", 78),
        ("_gRewkLB_EeqLtsTDMgEHBw", 15),
        ("__dDrANZLEeqiU8uzTY0Puw", 19),
        ("_jGNGMOCUEeu-FsE4WGMncQ", 3),
        ("_Ne8zEAATEeykFulkDFvkrg", 10),
        ("_I0nugeQUEe2jpIKbeHI0hw", 26),
    ],
)
def test_yield_diagram_elements(model, diag_uid, num_nodes):
    diagram_elements = sum(
        1 for _ in aird.yield_diagram_elements(model._loader, diag_uid)
    )

    assert diagram_elements == num_nodes
```

Note that this currently fails in two instances due to the aforementioned discrepancy; these are the values from `master` (assuming those are the correct ones).

Reply:
I copied your test and fixed the numbers. This test revealed the duplication bug and the need for special handling of ports. Great job!